### DISCUSSION TITLE
How to set a proportional bar width that scales nicely?

### DISCUSSION THREAD

**Discussion by quantoid** [vega/altair#2601] [Q&A] [answered]
Bar charts have a fairly narrow bar width by default, but if you configure a larger pixel width you get overlapping bars when users resize their browser window. What I need is to set the bar width to say 0.7 of the space between ticks, so they grow and shrink with the chart size and always have a gap  between them.

In Vega-lite you can do it [like this](https://vega.github.io/vega-lite/docs/bar.html#relative-bar-width):

    "mark": {"type": "bar", "width": {"band": 0.7}},

However, I can't find any way to do that in Altair, e.g. this throws an exception:

    Chart(data).mark_bar(width={"band": 0.7})

How do you specify proportional bar widths?

**Comment by mattijn** [comment, marked as answer]
Once the next version of Altair is released you can use `width=alt.RelativeBandSize(0.7)`. Using the main repo of altair this gives me currently: 
```python
import altair as alt
from vega_datasets import data

source = data.seattle_weather()

alt.Chart(source).mark_bar(
    width=alt.RelativeBandSize(0.7)
).encode(
    x='month(date):O',
    y='mean(precipitation):Q'    
)
```
![image](https://user-images.githubusercontent.com/5186265/167652028-536fe6af-1398-41d5-ae8a-53e39ab7f659.png)

**Reply by quantoid** [reply]
Thanks, is the next release imminent? Or many moons away?

**Reply by quantoid** [reply]
@jakevdp is there a best-guess timeframe for the next release with this included?

**Comment by joelostblom** [comment]
In addition to the answer above, you can also change the overall chart width, which will expand the bars. Using a step size indicates how many pixels you want to use for the width of each categorical axis value (ie each bar):

```python
import altair as alt
from vega_datasets import data

source = data.seattle_weather()

alt.Chart(source, width=alt.Step(30)).mark_bar().encode(
    x='month(date):O',
    y='mean(precipitation):Q'    
)
```

![image](https://user-images.githubusercontent.com/4560057/167709137-4d8ee7b0-ad1a-4b26-a3a6-72c682462220.png)

**Reply by quantoid** [reply]
Thanks, I considered that but unfortunately I'm using `streamlit.altair_chart` with `use_container_width=True` so the chart width varies.

**Reply by DanielFortuyn** [reply]
Was this ever resolved? I use streamlit too. When I update the data of a chart the bars remain the same size. it would be great if they could scale with the amount of x categories. The suggested solution does not work for me at the moment.

**Reply by joelostblom** [reply]
Does the accepted solution with `RelativeBandSize` not work for you?
